### PR TITLE
Permissionless Cage

### PR DESCRIPTION
**Pull request by hexonaut**
Additional safety mechanism which allows anyone to `cage` the adapter if a dangerous change comes through the admin timelock.

### CODE REVIEW

File: src/sushi.sol [hunk 1]
```diff
@@ -96,9 +139,51 @@ contract SushiJoin is CropJoin {
...
+            callData = abi.encodePacked(bytes4(keccak256(bytes(signature))), data);
+        }
+        require(callData.length >= 4, "SushiJoin/invalid-calldata");
+        bytes4 selector = bytes4(
+            (uint32(uint8(callData[0])) << 24) |
+            (uint32(uint8(callData[1])) << 16) |
+            (uint32(uint8(callData[2])) << 8) |
+            (uint32(uint8(callData[3])))
+        );
+        require(target == address(masterchef), "SushiJoin/wrong-target");
```
Comment (kmbarry1) [inline src/sushi.sol]: Why even have `target` as an argument? Seems like you can remove the argument and this check, and just use `masterchef` to get `txHash` below.

File: src/test/sushi-integration.t.sol [hunk 2]
```diff
@@ -574,4 +584,169 @@ contract SushiIntegrationTest is TestBase {
...
+        join.uncage();
+        assertTrue(join.live());
+        assertEq(pair.balanceOf(address(join)), 0);
+        assertEq(pair.balanceOf(address(masterchef)), tokensInMasterchef);
+
+        // Users can join/exit freely again
+        doJoin(user1, user1.getLPBalance());
+        doExit(user1, user1.getLPBalance());
+    }
+
```
Comment (kmbarry1) [inline src/test/sushi-integration.t.sol]:
```suggestion
```

**Review by kmbarry1 — COMMENTED** [review]
Generally looks good, had one minor suggestion for the queued transaction-based `cage`.

File: src/sushi.sol (hunk 1 above)
Comment (hexonaut) [reply src/sushi.sol]: Good call.

**Pull request merged**